hostscripts/clouddata/syncSLErepos

```diff
@@ -65,6 +65,20 @@ rsync_with_create() {
     $rsync "$src" "$dst"
 }
 
+rsync_until_repomd()
+{
+    local src="$1"
+    local dst="$2"
+
+    while
+        rsync_with_create $src $dst
+        [ $? = 24 ] && [ ! -r $dst/repodata/repomd.xml ]
```

Comment:
I'm not sure how this would fix the problem that @aspiers mentioned, it seems the problem isn't that repomd.xml is missing but that it's referring to a tarball that is missing:

```
> grep location /srv/www/suse-12.3/x86_64/repos/Cloud/repodata/repomd.xml
  <location href="repodata/57c4467ce18bb3b842c7c638392ea1a26c534c6ce00eeb97b222c9b92e73b3d1-appdata.xml.gz"/>
  <location href="repodata/ad8d7ff7108152713d90ef88a913a3382ae7b0935eb3a2d44023759f22d844ad-license.tar.gz"/>
  <location href="repodata/443319c4106cd73494406038585d050e1a3f8851dc309f182e47b6576416e5f0-filelists.xml.gz"/>
  <location href="repodata/39f493ee5adbdab0a2ed31df319e1bd54d5b14f14d68f459909dce776943dc0d-primary.xml.gz"/>
  <location href="repodata/76819c2eaea87cad2c97caf41e8ecf700b0da635cb8e04ade9cb6b6c214be2ba-susedata.xml.gz"/>
  <location href="repodata/df52963e3a3905a2e6973b7d0208654f0c6a7b2ca5c252a75b4c896adc83d3b9-other.xml.gz"/>
>ls /srv/www/suse-12.3/x86_64/repos/Cloud/repodata/
57c4467ce18bb3b842c7c638392ea1a26c534c6ce00eeb97b222c9b92e73b3d1-appdata.xml.gz  repomd.xml  repomd.xml.asc  repomd.xml.key
```

Reply:
The files seem to be there now. Would changing `&&` to `||` make more sense though?

Reply:
@cmurphy the problem is that rysync first generates a filelist to transfer, and then starts transfering. during the transfer, a new iso gets published, so it fails somewhere in the middle trying to sync the files (that it determined to sync beforehand). as the bandwidth to provo is insanely low, the window of a race is quite large.

Reply:
yeah, good point. not that the code would still be used... ;-) but good reminder from github that I forgot to delete it.